### PR TITLE
Add doc for default route related changes 

### PR DESCRIPTION
**Pull request by zjswhhh**
### Description of PR

Summary:
Fixes # (issue)

Realizing the default route related changes is actually affecting linkmgrd's state transition logics. Adding a doc for future reference. 

sign-off: Jing Zhang zhangjing@microsoft.com 

### Type of change

- [ ] Bug fix
- [ ] New feature
- [x] Doc/Design
- [ ] Unit test

### CODE REVIEW

File: doc/default_route.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+* Stop sending heartbeat   
+To shutdown heartbeat sending if default route is missing, is a straightforward way to fake an unhealthy status. A faked unhealthy status will help linkmgrd:
+    * keep the ToR in standby if one ToR is missing default route;
+    * avoid oscillation if both ToRs are missing defualt route (as eventually both will be `LinkProber:Wait` state).   
+
+Link prober will still be able to receive heartbeat. So if we remove & re-add the default route on both ToRs, a state transition path can be:   
+
+```
+         0.init state   ->  1.remove on A   ->  2.remove on B     -> 3.re-add on B      -> 4.re-add on A
+ToR A   active, healthy ->  standby, healthy->  active, unhealthy ->  standby, healthy  -> standby, healthy
```
Comment (yxieca) [inline doc/default_route.md:27]: Should 1 be standby, unhealthy?

Comment (yxieca) [reply doc/default_route.md:27]: Why would step 2 toggle when they both lost default gw?

Comment (zjswhhh) [reply doc/default_route.md:27]: It should be healthy. Though sending is suspended, it can still receive peer's HB.

Comment (yxieca) [reply doc/default_route.md:27]: Step 3 should also be standby unhealthy

Comment (zjswhhh) [reply doc/default_route.md:27]: > Why would step 2 toggle when they both lost default gw?

Addressed in section below ` Avoid proactive switches `.

Comment (zjswhhh) [reply doc/default_route.md:27]: > Step 3 should also be standby unhealthy

Same as Step 1.

**Review by lolyu — APPROVED** [review]

File: doc/default_route.md (hunk 1 above)
Comment (zjswhhh) [reply doc/default_route.md:27]: Discussed offline, resolving conversation.

**Pull request merged**